pygmt/tests/test_text.py

```diff
@@ -410,12 +410,17 @@ def test_text_nonstr_text():
     return fig
 
 
-@pytest.mark.mpl_image_compare
-def test_text_nonascii():
+@pytest.mark.mpl_image_compare(filename="test_text_nonascii.png")
+@pytest.mark.parametrize("encoding", ["ISOLatin1+", "Standard+"])
+def test_text_nonascii(encoding):
     """
     Test passing text strings with non-ascii characters.
+
+    Default PS_CHAR_ENCODING setting should not affect the result.
     """
     fig = Figure()
+    if encoding == "Standard+":  # Temporarily set the PS_CHAR_ENCODING to "Standard+".
```

Comment:
Ideally, we should test the case that PS_CHAR_ENCODING is set to "Standard+" in a global "gmt.conf" file. However, it's technically difficult. Similar to the test `test_gmt_compat_6_is_applied ` at https://github.com/GenericMappingTools/pygmt/blob/763dc07931ebe597a4466ff30717195023e50f11/pygmt/tests/test_session_management.py#L31

Generally we need to:

1. Generate a gmt.conf file in the current directory
2. Kill the global session
3. Start a new session
4. Create a new figure and adding non-ASCII characters to it
5. End the session so the "gmt.conf" file won't affect other tests
6. Start a new, clean global session

But the issue is that, after starting a new global session, we can't return the Figure instance for comparing images, because `mpl_image_compare` needs to call `Figure.savefig` which no longer works after the session is destroied.

So here we only test the case that `PS_CHAR_ENCODING` is changed in the middle of a script.